### DISCUSSION TITLE
Transform claim key not found to an empty string

### DISCUSSION THREAD

**Discussion by guilhermebentomarques** [ThreeMammals/Ocelot#1851] [Q&A]
Hi Ocelot team,

It's a question, not an issue.

Is it possible to check if a key exists and if not pass an empty string?

For example:

What I have: 
`"AddQueriesToRequest": {
        "userId": "Claims[sub] > value"
}`

Error Code: CannotFindClaimError Message: 
Cannot find claim for key: sub errors found in ResponderMiddleware. Setting error response for request

What I would like to do (If the key doesn't exist, send an empty string.)
`"AddQueriesToRequest": {
        "userId": "Claims[sub] ? Claims[sub] > value : '' "
}`

Thank you.

**Comment by ks1990cn** [comment]
@guilhermebentomarques  No! this breaks functionality. Why you are trying to put claim key after being authenticated? Its not a right idea.

**Comment by raman-m** [comment]
Hello, Guilherme!
Welcome to Ocelot world!

-----
> Error Code: CannotFindClaimError Message:
Cannot find claim for key: sub errors found in ResponderMiddleware. Setting error response for request

Why not to develop custom [Delegating Handler](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html) module?
Why not to develop custom [Authorization middleware](https://ocelot.readthedocs.io/en/latest/features/authorization.html) or [other middlewares](https://ocelot.readthedocs.io/en/latest/features/middlewareinjection.html) as additional step to process the claims?

-----
> What I would like to do (If the key doesn't exist, send an empty string.)
> `"AddQueriesToRequest": {
>         "userId": "Claims[sub] ? Claims[sub] > value : '' "
> }`
> 

Ocelot doesn't support this syntax. We need to develop a feature.
We welcome any contributions from you.
